examples/platform/silabs/efr32/wf200/host_if.cpp

```diff
@@ -534,9 +542,37 @@ static void wfx_events_task(void * p_arg)
         {
             if (!retryInProgress)
             {
+                retryInProgress = true;
+                if (!is_disconnection_event)
+                {
+                    /* After the reboot or a commissioning time device failed to connect with AP.
+                     * Device will retry to connect with AP upto WFX_RSI_CONFIG_MAX_JOIN retries.
+                     */
+                    SILABS_LOG("%s: Next attempt after %d sec", __func__, CONVERT_MS_TO_SEC(WLAN_RETRY_TIMER_MS));
+                    if (retryJoin < MAX_JOIN_RETRIES_COUNT)
+                        vTaskDelay(WLAN_RETRY_TIMER_MS);
+                }
+                else
+                {
+                    /* After disconnection
+                     * At the telescopic time interval device try to reconnect with AP, upto WLAN_MAX_RETRY_TIMER_MS
+                     * intervals are telescopic. If interval exceed WLAN_MAX_RETRY_TIMER_MS then it will try to
+                     * reconnect at WLAN_MAX_RETRY_TIMER_MS intervals.
+                     */
+                    if (retryInterval > WLAN_MAX_RETRY_TIMER_MS)
+                    {
+                        retryInterval = WLAN_MAX_RETRY_TIMER_MS;
+                        SILABS_LOG("%s: Next attempt after %d Seconds", __func__, CONVERT_MS_TO_SEC(WLAN_MAX_RETRY_TIMER_MS));
+                    }
+                    else
+                    {
+                        SILABS_LOG("%s: Next attempt after %d Seconds", __func__, CONVERT_MS_TO_SEC(retryInterval));
+                    }
+                    vTaskDelay(pdMS_TO_TICKS(retryInterval));
+                    retryInterval += retryInterval;
+                }
```

Comment:
Pattern#3

Reply:
Pattern #1 and #2 are slight different whereas for pattern #2 and #3 are same but used in rs9116 and wf200 resp so only one file will get complied at a time.